core/src/main/java/org/apache/james/core/StringWrapper.java

```diff
@@ -0,0 +1,57 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core;
+
+import java.util.Objects;
+
+import com.google.common.base.MoreObjects;
+
+public class StringWrapper {
```

Comment:
From what I can see: `QuotaComponent` still can work without extending `StringWrapper`.

Q: What is the point of introducing this  `StringWrapper`?

Reply:
to reuse some code in other classes: QuotaType, QuotaScope

Reply:
If you mean "reuse some code" is reuse the String attribute then I am not convinced.

Why just not declare that string attribute directly in `QuotaType`, `QuotaScope`, and `QuotaComponent`?

IMO the types of classes `QuotaType`, `QuotaScope`, and `QuotaComponent` have nothing in common (imagine in the real world) that needs inheritance. String is just an implementation detail of those class, extending all of them from `StringWrapper` just add more complexity and potentially misunderstand binding IMO.